# Activity - Multivariate LR 02

In [1]:
import pandas as pd
import numpy as np
from sklearn import linear_model

In [3]:
df = pd.read_csv('./data/multi_var_data.csv')
df

Car  Volume  Weight  CO2
0       Toyoty    1000     790   99
1   Mitsubishi    1200    1160   95
2        Skoda    1000     929   95
3         Fiat     900     865   90
4         Mini    1500    1140  105
5           VW    1000     929  105
6        Skoda    1400    1109   90
7     Mercedes    1500    1365   92
8         Ford    1500    1112   98
9         Audi    1600    1150   99
10     Hyundai    1100     980   99
11      Suzuki    1300     990  101
12        Ford    1000    1112   99
13       Honda    1600    1252   94
14      Hundai    1600    1326   97
15        Opel    1600    1330   97
16         BMW    1600    1365   99
17       Mazda    2200    1280  104
18       Skoda    1600    1119  104
19        Ford    2000    1328  105
20        Ford    1600    1584   94
21        Opel    2000    1428   99
22    Mercedes    2100    1365   99
23       Skoda    1600    1415   99
24       Volvo    2000    1415   99
25    Mercedes    1500    1465  102
26        Audi    2000    1490  104
27        Audi    2000    1725  114
28       Volvo    1600    1523  109
29         BMW    2000    1705  114
30    Mercedes    2100    1605  115
31       Volvo    2000    1746  117
32        Ford    1600    1235  104
33         BMW    1600    1390  108
34        Opel    1600    1405  109
35    Mercedes    2500    1395  120

## Train & Predict

In [4]:
reg = linear_model.LinearRegression()
reg.fit(df[['Volume','Weight']], df['CO2'])

LinearRegression()

In [5]:
reg.predict([[1300, 2300]])

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([107.2087328])

In [6]:
reg.predict([[1100, 1300]])

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([98.09673402])

## Verify

In [7]:
reg.coef_

array([0.00780526, 0.00755095])

In [8]:
reg.intercept_

79.6947192911594

In [11]:
# weight is 2300kg, and the volume is 1300cm3
0.00780526*1300 + 0.00755095*2300 + 79.6947192911594

107.2087422911594

In [12]:
# weight is 1300kg, and the volume is 1100cm3
0.00755095*1100 + 0.00780526*1300 + 79.6947192911594

98.1476022911594

1. What will be the CO2 emission of a car where the weight is 2300kg, and the volume is 1300cm3? **107.2087328**


2. What will be the CO2 emission of a car where the weight is 1300kg, and the volume is 1100cm3? **98.09673402**